src/lib/rest/ConnectionInfo.h

```diff
@@ -168,6 +168,7 @@ class ConnectionInfo
   RequestType                requestType;  // FIXME P2: To Be Removed (found inside restServiceP->request (restServiceP->type))
   std::string                acceptHeaderError;
   struct timeval             transactionStart;  // For metrics
+  std::string                entityIdFromUrlPath;
```

Comment:
Why is this needed?

Reply:
This is meant for requests of the type GET /v2/entities/<entityId>

Reply:
NTC

Reply:
I understand this parameter is used in queryForward() in

```
    if (ciP->entityIdFromUrlPath != "")
    {
      extraParams = "&id=" + ciP->entityIdFromUrlPath;
    }
```

However, the entity id should be taken from the QueryContextRequest which is passed as argument to the function. Note that the service routing logic would have transformed the NGSIv2 request in the equivalent QueryContextRequest and the id should be there.

Moreover, maybe this should be prepared for multiple IDs (i.e. `id=E1,E2`). The QueryContextRequest could have several entity ids, e.g. `POST /v2/op/query` or `POST /v1/queryContext` with several entities in the payload.

Reply:
Also related with the usage of QueryContextRequest parameter in queryForward() as source of entity ids, the following should be removed:

```
    if (ciP->uriParam["id"] != "")
    {
      extraParams = "&id=" + ciP->uriParam["id"];
    }
```

Reply:
Found another code smell with extraParams (`extraParams =` instead of `extraParams +=`).
Fixed in 6486136a8

Reply:
Good for the changes in 6486136, but in addition I understand that `ciP->entityIdFromUrlPath = entityId;` should be removed in getEntity.cpp

Reply:
Fixed in 7d1cb1056